sdk/trace/simple_span_processor.go

```diff
@@ -91,7 +91,17 @@ func (ssp *simpleSpanProcessor) Shutdown(ctx context.Context) error {
 		select {
 		case err = <-done:
 		case <-ctx.Done():
-			err = ctx.Err()
+			// It is possible for the exporter to have immediately shut down
+			// and the context to already be done. In that case this select
+			// statement will randomly choose a case. This could result in a
+			// different response for similar scenarios. Instead, double check
+			// if the exporter shut down at the same time and return that
+			// error if so.
+			if expErr, ok := <-done; ok {
+				err = expErr
+			} else {
+				err = ctx.Err()
+			}
```

Comment:
I am not sure if that is what you wanted to do after reading https://github.com/open-telemetry/opentelemetry-go/pull/2290#discussion_r731046615

1. Line 103 would rather be never executed as I do not see any `close(done)`
2. It will always wait on `<- done` even if cancelation occurs. We are not exiting until `exp.Shutdown(ctx)` finishes

I am guessing that you may want to do something like this instead:

```go
select {
case expErr :=<- done:
  err =  expErr
default:
  err = ctx.Err()
}
```

Personally, I was more familiar with the following pattern:

```go
go shutdown()
select {
case err = <-done:
    return // here we make sure that `<-done:` has higher priority than context error
default:
}
select {
case err = <-done:
case <-ctx.Done():
    err = ctx.Err()
}
```

Reply:
_Agree - I edited the comment before I saw your reply_